pydantic/_internal/_model_construction.py

```diff
@@ -75,6 +74,13 @@ def inspect_namespace(  # noqa C901
     for var_name, value in list(namespace.items()):
         if var_name == 'model_config':
             continue
+        elif (
+            isinstance(value, type)
+            and value.__module__ == namespace['__module__']
+            and value.__qualname__.startswith(namespace['__qualname__'])
+        ):
+            # `value` is a nested type defined in this namespace; don't error
```

Comment:
I'm missing something. Conceptually, why is "skipping" nested types defined in this namespace a solution for this issue?

Reply:
Without this skipping, if you do:
```python
class A(BaseModel):
    class B:
        x = 1
```
it will error. But if you are defining `B` inside of `A`, I'm sort of assuming that was intentional. (This is also necessary to allow `class Config` to not error, though we could handle that based on the name `Config`. I just think we should also allow nested types to be defined without erroring.)

Reply:
got it, thanks 🙏